### ISSUE TITLE
Debian/Ubuntu (deb) installation failes with v1.12.0 installer

### ISSUE THREAD

**Issue by AndreasAmMueller** [vernemq/vernemq#1818]
### Environment

- VerneMQ Version: 1.12.0
- OS: Ubuntu 20.04.2 LTS / Debian 10 (Buster)
- Architecture: amd64

### Expected behaviour

After installing the dependencies the `dpkg -i` command should run without any error.

### Actual behaviour

The installer breaks due to a malformed post-install script, see below

```
root@localhost:~# wget https://github.com/vernemq/vernemq/releases/download/1.12.0/vernemq-1.12.0.buster.x86_64.deb
[...]
root@localhost:~# dpkg -i vernemq-1.12.0.buster.x86_64.deb
Selecting previously unselected package vernemq.
(Reading database ... 9317 files and directories currently installed.)
Preparing to unpack vernemq-1.12.0.buster.x86_64.deb ...
Unpacking vernemq (1621516682:1.12.0) ...
Setting up vernemq (1621516682:1.12.0) ...
/var/lib/dpkg/info/vernemq.postinst: 11: /var/lib/dpkg/info/vernemq.postinst: Syntax error: "else" unexpected
dpkg: error processing package vernemq (--install):
 installed vernemq package post-installation script subprocess returned error exit status 2
Errors were encountered while processing:
 vernemq
```

**Comment by ioolkos** [comment]
@AndreasAmMueller thanks, jep, I've seen the same thing. Verne is probably installed and you can start it, but maybe that postinst will pop up on every dpkg call, or something.
Let me know if you have an idea, I need to investigate it.

---
https://github.com/sponsors/vernemq

**Comment by ioolkos** [comment]
I see that other projects have similar issues. Just one example: https://github.com/syncthing/syncthing/issues/7548
Reading that, it could be related to FPM (used here too in the internal building of packages).

---
https://github.com/sponsors/vernemq

**Comment by ioolkos** [comment]
Seems this is the issue/fix for .deb based systems: https://github.com/jordansissel/fpm/pull/1752/files but that PR is not yet merged in FPM

---
https://github.com/sponsors/vernemq

**Comment by AndreasAmMueller** [comment]
Ok, you're faster than me.

Had seen that fix too on my search... so we'll wait for that pull-request to be done :-)

**Comment by ioolkos** [comment]
@AndreasAmMueller yeah, thanks. It's related to internal package building for the download packages (with the required subscription). So internal fixing needed. Still weird, it must be related to switching to Buster or something.

---
https://github.com/sponsors/vernemq

**Comment by ioolkos** [comment]
Ok, in `/var/lib/dpkg/info/vernemq.postinst`: "if then else" clearly is not valid syntax. We don't say what to do (then) if `systemctl` is not enabled for Verne. Maybe we just should do nothing.


```bash
#!/bin/sh


after_upgrade() {
    :

systemctl --system daemon-reload >/dev/null || true
debsystemctl=$(command -v deb-systemd-invoke || echo systemctl)
if ! systemctl is-enabled vernemq >/dev/null 
then
else
    $debsystemctl restart vernemq >/dev/null || true
fi
}
```
Maybe we can adapt those lines, and force-install with dpkg as a workaround.

**Comment by JacoFourie** [comment]
Hi. I have the same issue. How do we fix this? 
The status says it is installed but half-configured and it wont start up.

**Comment by AndreasAmMueller** [comment]
At the moment you can fix the broken file yourself or just install the older version (latest before 1.12.0 => 1.11.0)

**Comment by JacoFourie** [comment]
> At the moment you can fix the broken file yourself or just install the older version (latest before 1.12.0 => 1.11.0)

Where do I get the 1.11.0 version?

**Comment by AndreasAmMueller** [comment]
> 
> 
> > At the moment you can fix the broken file yourself or just install the older version (latest before 1.12.0 => 1.11.0)
> 
> Where do I get the 1.11.0 version?

https://vernemq.com/downloads/archived.html

or using Docker, you need to append the version tag `vernemq/vernemq:1.11.0`

**Comment by JacoFourie** [comment]
Thank you. I downloaded version 1.11 but it wont install also. It gives an error that it needs 

dpkg: dependency problems prevent configuration of vernemq:
 vernemq depends on libssl1.0.0 (>= 1.0.1); however:
  Package libssl1.0.0 is not installed.

And when I try to install that lib I get this error.

sudo apt-get install -y libssl1.0.0
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package libssl1.0.0 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'libssl1.0.0' has no installation candidate


The lib was dropped from Ubuntu. You now have to use libssl1.1

**Comment by JacoFourie** [comment]
OK so I managed to get the broker running using Docker. I have not used Docker in the past as I am more a developer than an system admin, so now I have to figure out how to setup the configuration of the broker inside Docker to use the DB for ACL etc.
.

**Comment by ioolkos** [comment]
@JacoFourie I think for the borked install, you need to manually edit that first wrong "if then" in the `/var/lib/dpkg/info/vernemq.postinst` script to something like the following (only first lines pasted):

```sh
#!/bin/sh

after_upgrade() {
    :

systemctl --system daemon-reload >/dev/null || true
debsystemctl=$(command -v deb-systemd-invoke || echo systemctl)
if systemctl is-enabled vernemq >/dev/null; then
    $debsystemctl restart vernemq >/dev/null || true
fi
}
```
And then hopefully running
`sudo dpkg --configure vernemq` will fix the install.

@AndreasAmMueller thanks for your comments & let me know should you have a better way.

---
:point_right: Thank you for supporting VerneMQ: https://github.com/sponsors/vernemq

**Comment by JacoFourie** [comment]
> sudo dpkg --configure vernemq

Thank you. That fixed the install. I got the broker working. Now to figure out the rest.

**Comment by Fernandoaml** [comment]
> sudo dpkg --configure vernemq

This is hopeful a lot. Fixed the issue of installation. Thanks a lot!

**Comment by ioolkos** [comment]
Fixed in 1.12.2

**Comment by syedzahidulla** [comment]
> 

Iam getting same error..
Iam new to u ubuntu can you describe what next steps to be done.

Thankyou
